source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -912,6 +950,35 @@ void ShaderGraph::validateNames(ShaderGenerator& shadergen)
     }
 }
 
+void ShaderGraph::populateInputColorTransformMap(const Node& node, ShaderNodePtr shaderNode, ValueElementPtr input, const string& targetColorSpace)
+{
+    ShaderInput* shaderInput = shaderNode->getInput(input->getName());
+    const string& sourceColorSpace = input->getAttribute(Element::COLOR_SPACE_ATTRIBUTE);
+    if (!sourceColorSpace.empty())
+    {
+        // Can skip inputs with connections as they are not legally allowed to have colorspaces specified.
+        if (!shaderInput->connection)
+        {
+            if(shaderInput->type == Type::COLOR3 || shaderInput->type == Type::COLOR4)
+            {
+                // If we're converting between two identical color spaces than we have no work to do.
+                if (sourceColorSpace != targetColorSpace)
+                {
+                    ColorSpaceTransform transform( sourceColorSpace, targetColorSpace, *shaderInput->type );
+                    _inputColorTransformMap.emplace(shaderInput, transform);
+                }
+            }
+            else if(shaderInput->type != Type::FILENAME)
+            {
+                throw ExceptionShaderGenError("Color space attribute for: '" + node.getName() + "." + input->getName() + "' of unsupported type (must be color3 or color4).");
+            }
+        }
+        else
+        {
+            throw ExceptionShaderGenError("Color space attribute for: '" + node.getName() + "." + input->getName() + "' invalid as connection exists.");
```

Comment:
Note sure if this should throw or just silently ignore the colorspace attribute. The colorspace attribute on an input is really tied to the value attribute of the input, and it's valid to have a value while being connected. So I'm leaning towards allowing a colorspace attribute but just ignoring it for the connected case.